etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/messagelink/MessageLink.java

```diff
@@ -0,0 +1,53 @@
+package gov.hhs.cdc.trustedintermediary.etor.messagelink;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * This class represents a link between messages. Each link has a unique ID and is associated with a
+ * set of message IDs to link.
+ */
+public final class MessageLink {
+    private UUID linkId;
+    private Set<String> messageIds;
+
+    public MessageLink() {
+        this.messageIds = new HashSet<>();
+    }
+
+    public MessageLink(UUID linkId, String messageId) {
+        this.linkId = linkId;
+        this.messageIds = new HashSet<>(Collections.singleton(messageId));
+    }
+
+    public MessageLink(UUID linkId, Set<String> messageIds) {
+        this.linkId = linkId;
+        this.messageIds = new HashSet<>(messageIds);
+    }
+
+    public void setLinkId(UUID linkId) {
+        this.linkId = linkId;
+    }
+
+    public UUID getLinkId() {
+        return linkId;
+    }
+
+    public void setMessageIds(Set<String> messageIds) {
+        this.messageIds = new HashSet<>(messageIds);
+    }
+
+    public Set<String> getMessageIds() {
+        return Collections.unmodifiableSet(messageIds);
```

Comment:
You've put some protections in place in this class to defend against modifications.  For example, above use the copy constructor to get your own copy of the messageIds.  Here, you return a new version of the set that is unmodifiable.

Was there a reason for this?  I don't have an opinion, but I'm curious what you're thinking was.

Reply:
I based this off of an example. I was just trying to handle Sets in Java, which I don't have much experience with so I'm not very familiar with the syntax. Now that you mention it, I don't think it's really necessary